terminus-ui/cohort-date-range/src/cohort-date-range.component.ts

```diff
@@ -0,0 +1,152 @@
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+  ViewEncapsulation,
+} from '@angular/core';
+import { FormBuilder } from '@angular/forms';
+
+/**
+ * Represents date cohort object that passed in
+ */
+export interface TsDateCohort {
+  display: string;
+  range: {
+    startDate: string | Date;
+    endDate: string | Date;
+  };
+}
+
+/**
+ * Event object emitted by TsCohortDateRange when there is date range change
+ */
+
+export class TsCohortDateRangeChanged {
+  constructor(
+    public start: string | Date,
+    public end: string | Date,
+    public source: TsCohortDateRangeComponent,
+  ) { }
+}
+
+/**
+ * This is the cohort-date-range UI Component
+ *
+ *
+ * @example
+ * <ts-cohort-date-range
+ *              [allowCustomDates]="true"
+ *              [cohorts]="myCohorts"
+ *              (cohortDateRangeChange)="myFunc($event)"
+ *              <ts-date-range></ts-date-range>
+ *              <ts-select></ts-select>
+ * ></ts-cohort-date-range>
+ *
+ * <example-url>https://getterminus.github.io/ui-demos-release/components/cohort-date-range</example-url>
+ */
+@Component({
+  selector: 'ts-cohort-date-range',
+  templateUrl: './cohort-date-range.component.html',
+  styleUrls: ['./cohort-date-range.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  exportAs: 'tsCohortDateRange',
+  host: {
+    'class': 'ts-cohort-date-range',
+    '[class.ts-cohort-date-range-disabled]': 'isDisabled',
+    '[attr.disabled]': 'isDisabled',
+    '[attr.aria-disabled]': 'isDisabled',
+  },
+})
+export class TsCohortDateRangeComponent {
+  public disableDateRange = false;
+  /**
+   * Define whether custom date is allowed
+   */
+  @Input()
+  public allowCustomDates = true;
+
+  /**
+   * Cohort dates
+   */
+  @Input()
+  public cohorts!: ReadonlyArray<TsDateCohort>;
+
+  /**
+   * Disable the component
+   */
+  @Input()
+  public isDisabled = false;
+
+  /**
+   * Cohort change event emitter
+   */
+  @Output()
+  public readonly cohortDateRangeChanged: EventEmitter<TsCohortDateRangeChanged> = new EventEmitter();
+
+  /**
+   * Initialize the date range with empty start and end date
+   */
+  public formGroup = this.formBuilder.group({
+    dateRange: this.formBuilder.group({
+      startDate: [null],
+      endDate: [null],
+    }),
+  });
+
+  /**
+   * Get reference of dateRange from formGroup
+   */
+  public dateRangeFg = this.formGroup.get('dateRange');
+
+  constructor(
+    public formBuilder: FormBuilder,
+  ) { }
+
+  /**
+   * Called when cohort date range changed and emit cohortDaterangeChanged event
+   *
+   * @param event - triggered by date range change
+   */
+
+  public cohortDateRangeChange(event): void {
+    this.cohortDateRangeChanged.emit(new TsCohortDateRangeChanged(event.start, event.end, this));
+  }
+
+  /**
+   * Called when selection changed
+   *
+   * @param event - TsSelectChangeEvent
+   */
+
+  public selectionChange(event) {
```

Comment:
Unneeded line break (pretty sure our docs generator won't see that these items are related if an extra line break exists)

Reply:
Also missing return type